javascript/lang/hardcoded_secret.yml

```diff
@@ -21,6 +21,63 @@ patterns:
       - variable: STRING_LITERAL
         detection: string_literal
         contains: false
+  - pattern: crypto.$<METHOD>($<_>, $<STRING_LITERAL>$<...>)
+    filters:
+      - variable: METHOD
+        values:
+          - createCipher
+          - createDecipher
+          - createCipheriv
+          - createDecipheriv
+          - createHmac
+      - variable: STRING_LITERAL
+        detection: string_literal
+        contains: false
```

Comment:
Could you confirm what the `contains` property does, I suppose it says the variable shouldn't be of type `string _literal` (can be anything else except the string_literal).

Reply:
`contains` specifies whether the rule must match at the variable's node only (false), or whether any node in it's subtree can be used (true). The default is true. 

Here we need to use that so `"a" + b` doesn't match the string literal `"a"`. We're saying the entire expression must be a string literal.

Reply:
For me the naming is very confusing, can we rename it to `submatches`, `subtreeMatches`, or something else that is more explicit?